src/avram/upsert.cr

```diff
@@ -90,6 +90,27 @@ module Avram::Upsert
     end
   end
 
+  macro upsert_unique_on(*attribute_names)
+    def self.upsert(upserts : Array(X)) forall X
+      \{%
+        if X.union?
```

Comment:
Should another check/error be added in case `X` isn't a `NamedTuple`? Only if the error would be more helpful than the current default. The double splat below probably forces it to the type anyways

Reply:
Yeah, double splat will catch it.

Reply:
added better errors 👍